src/layout/size.rs

```diff
@@ -16,10 +16,28 @@ pub struct Size {
 }
 
 impl Size {
+    /// A zero sized Size
+    pub const ZERO: Self = Self::new(0, 0);
+
     /// Create a new `Size` struct
     pub const fn new(width: u16, height: u16) -> Self {
         Self { width, height }
     }
+
+    /// Creates a [`Rect`] from this size at the origin (0,0).
+    pub const fn at_origin(self) -> Rect {
+        self.with_position(Position::ORIGIN)
+    }
```

Comment:
+1 for having methods that take a size and produce a Rect, but the naming feels subjectively odd. Would it make sense to create these on Rect instead? E.g. `Rect::at_position(size: Size)` / `Rect::at_origin(size: Size)`?

Reply:
#1254 uses `impl From<(Position, Size)> for Rect` now. Its less readable but works.